crates/op-rbuilder/src/payload_builder_vanilla.rs

```diff
@@ -421,14 +445,34 @@ impl<Txs> OpBuilder<'_, Txs> {
             .builder_signer()
             .map_or(0, |_| estimate_gas_for_builder_tx(message.clone()));
         let block_gas_limit = ctx.block_gas_limit() - builder_tx_gas;
+        // Save some space in the block_da_limit for builder tx
+        let builder_tx_da_size = ctx
+            .estimate_builder_tx_da_size(state, builder_tx_gas, message.clone())
+            .unwrap_or(0);
+        let block_da_limit = ctx
+            .da_config
+            .max_da_block_size()
+            .map(|da_size| da_size - builder_tx_da_size as u64);
+        // Check that it's possible to create builder tx, considering max_da_tx_size, otherwise panic
+        if let Some(tx_da_limit) = ctx.da_config.max_da_tx_size() {
+            // Panic indicate max_da_tx_size misconfiguration
+            assert!(tx_da_limit >= builder_tx_da_size as u64);
```

Comment:
is this copied from the payload builder in reth? wondering if there should be an error log as well here

Reply:
No, this little piece is mine
I'll add a message, thank you!